### PR TITLE
 #203 rework make targets naming

### PR DESCRIPTION
**Pull request by Shr3ps**

### CODE REVIEW

**Review by xp-1000 — COMMENTED** [review]
why not add an alias

File: Makefile [hunk 1]
```diff
@@ -70,3 +70,10 @@ stack:
 .PHONY: module
 module: 
 	./scripts/module/bootstrap.sh $(COMMAND_ARGS)
+
+.PHONY: module-update
+module-update:
+	./scripts/module/gen_detectors.sh $(COMMAND_ARGS)
+	./scripts/module/gen_outputs.sh $(COMMAND_ARGS)
+	./scripts/module/lint.sh $(COMMAND_ARGS)
+	CI=true ./scripts/module/gen_doc.sh $(COMMAND_ARGS)
```
Comment (xp-1000) [inline Makefile:79]:
```suggestion
module-update: modules lint
```

**Review by xp-1000 — CHANGES_REQUESTED** [review]

**Review by xp-1000 — APPROVED** [review]

**Comment by xp-1000** [comment]
@Shr3ps I pushed a couple of changes on Makefile (and updated the related doc: https://github.com/claranet/terraform-signalfx-detectors/blob/203_makefile_improve/docs/development.md#tldr) which should cover what you need in addition to make the usage easier (I hope).

what do you think?

**Review by xp-1000 — APPROVED** [review]

**Comment by xp-1000** [comment]
- all targets now start with an "action" (init, check, update..)
- targets are now long (but bash completion is installed on the dev env)
- it produces an human understanding sentence (update module doc)
- targets for deadlinks and spell checks added
- the special target update-module add the check-module one to get a
  full ready PR
- CI and doc updated

PR ready for reviews

**Review by jcabeza — APPROVED** [review]

**Review by Gati0 — APPROVED** [review]

File: docs/development.md [hunk 2]
```diff
@@ -37,45 +39,99 @@ rules and templates homogeneous to all modules.
...
+Most of contributions are related to [detectors change type](#detectors) and do no require 
+fully understand and know all possible targets, scripts and check.
+
+The essential things to know are:
+
+* to understand lot of the files are auto generated and checked by the CI so you have to 
+respect this.
+* to never edit a `*-gen*` file or the `README.md` in modules, update its underlying yaml 
+configuration files in `conf/` directory of the module.
+* as soon as you make a change on a module, run `make update-module` command.
```
Comment (Shr3ps) [inline docs/development.md:65]:
```suggestion
* as soon as you make any change on a module, run `make update-module` command to update every generated files.
```

File: docs/development.md [hunk 3]
```diff
@@ -37,45 +39,99 @@ rules and templates homogeneous to all modules.
...
+
+A common contribution which cover all required targets for most of detectors related changes 
+is the creation of a new module of detectors:
 
-- `dev`: the default and optional target run the container ready to run other commands
+```bash
+$ echo "Set required input information as environment variables"
+$
+$ SOURCE_TYPE="smart-agent" # or "internal", "integration", "organization", "otel-collector"
+$ MODULE_NAME="my-new-module" # use comma separated string to name your module
```
Comment (Shr3ps) [inline docs/development.md:76]:
```suggestion
$ MODULE_NAME="my-new-module" # use dash separated word to name your module
```

File: docs/development.md [hunk 4]
```diff
@@ -37,45 +39,99 @@ rules and templates homogeneous to all modules.
...
+$
+$ echo "Init a new module creating ready directory to welcome detectors"
+$ make init-module ${MODULE_DIR} # same as "make init-module ${SOURCE_TYPE} ${MODULE_NAME}"
+$
+$ echo "Create a new detector using the heartbeat sample configuration and updating module and metric"
+$ cp scripts/templates/examples/heartbeat-simple.yaml modules/${MODULE_DIR}/conf/00-heartbeat.yaml
+$ sed -i "s/^\(module:\).*$/\1 ${MODULE_NAME}/" modules/${MODULE_DIR}/conf/00-heartbeat.yaml
+$ sed -i "s/^\([[:space:]]*metric:\).*$/\1 \"${METRIC_NAME}\"/" modules/${MODULE_DIR}/conf/00-heartbeat.yaml
+$ 
+$ echo "Full update the module (doc, code, add/remove/change detectors ..)"
```
Comment (Shr3ps) [inline docs/development.md:88]:
```suggestion
$ echo "Full update the module (doc, terraform code, add/remove/change detectors...)"
```

File: docs/development.md [hunk 5]
```diff
@@ -37,45 +39,99 @@ rules and templates homogeneous to all modules.
...
+$ make update-module ${MODULE_DIR}
+```
+
+That is all! and from now you will have to run `make update-module` before to push any change, it 
+will take care to regenerate code, documentation and even check the module to be sure to pass the CI.
```
Comment (Shr3ps) [inline docs/development.md:93]:
```suggestion
will take care of regenerating code, documentation and even check and validate the module to pass the CI.
```

File: docs/development.md [hunk 6]
```diff
@@ -37,45 +39,99 @@ rules and templates homogeneous to all modules.
...
+
+If this is not enough for your change or if you want to fully understand all capabilities and dependencies 
+you will have to read the rest of this documentation.
+
+## Make targets
+
+You seen above the `update-module` and `init-module` `make` targets which are very common
```
Comment (Shr3ps) [inline docs/development.md:100]:
```suggestion
You have seen above the `update-module` and `init-module` `make` targets which are very common 
```

**Pull request merged**